### DISCUSSION TITLE
DTI tensor fit from the b-matrix instead of bvecs.

### DISCUSSION THREAD

**Discussion by Pedro-Filipe** [dipy/dipy#3411] [Reconstruction]
Hello,

I've been using DiPy to fit a diffusion tensor with code like this:
```python
gtab = gradient_table(bvals, bvecs)
tenmodel = dti.TensorModel(gtab)
tenfit = tenmodel.fit(maskdata)
```

But now, I have some Bruker data in DICOM format, where each image has a `bval` and `bmatrix`, instead of `bval` and `bvec`.

My question is: is there a way to get `gtab` with `bval` and `bmatrix`? 

Thank you.

---

DICOM header sample for one image:
```text
{'MRDiffusionSequence': [{'DiffusionDirectionality': 'BMATRIX',
                          'DiffusionBValue': 95.71229490959814,
                          'DiffusionBMatrixSequence': [{'DiffusionBValueXX': 4.264692682413525,
                                                        'DiffusionBValueXY': 18.785912996304578,
                                                        'DiffusionBValueXZ': 3.8382234141721665,
                                                        'DiffusionBValueYY': 87.99320115442967,
                                                        'DiffusionBValueYZ': 16.907321696674096,
                                                        'DiffusionBValueZZ': 3.454401072754944}]}],
```

**Comment by skoudoro** [comment]
Hi @Pedro-Filipe, 

I would recommend to use https://github.com/rordenlab/dcm2niix. 

it will transform your DICOM to NIFTI in a minute and then you will be able to have to gtab with `bval` and `bvec`

**Reply by Pedro-Filipe** [reply]
Thank you. 

I have actually try this and it works, but I was hoping to use the bmatrix values directly in DiPy. Not sure if possible.
